LINDSAY ARNEZ - EJERCICIO CLINICA.cpp

```diff
@@ -0,0 +1,141 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Paciente {
+protected:
+    string nombre;
+    int    edad;
+public:
+    Paciente(string _nombre, int _edad) : nombre(_nombre), edad(_edad) {
+        this->nombre = _nombre;
+        this->edad = _edad;
+    }
+
+    string getNombre() { 
+        return nombre; 
+    }
+    int getEdad() {
+        return edad; 
+    }
+
+    virtual double calcularDescuento() = 0;
+
+    virtual void mostrarDatos() {
+        cout << "Nombre: " << nombre << ", Edad: " << edad;
+    }
+};
+
+class Externo : public Paciente {
+public:
+    Externo(string _nombre, int _edad) : Paciente(_nombre, _edad) {}
+
+    double calcularDescuento() {
+        if (edad > 80) return 0.50;
+        if (edad > 65) return 0.20;
+        return 0.0;
+    }
+
+    void mostrarDatos() {
+        cout << "==== Externo ==== ";
+        Paciente::mostrarDatos();
+        cout << ", Descuento: " << calcularDescuento() * 100 << "%\n";
+    }
+};
+
+class Asegurado : public Paciente {
+public:
+    Asegurado(string _nombre, int _edad) : Paciente(_nombre, _edad) {}
+
+    double calcularDescuento() {
+        if (edad < 5 || edad > 50) return 0.80;
+        return 0.50;
+    }
+
+    void mostrarDatos() {
+        cout << "=== Asegurado === ";
+        Paciente::mostrarDatos();
+        cout << ", Descuento: " << calcularDescuento() * 100 << "%\n";
+    }
+};
+
+class Atencion {
+protected:
+    string fecha;  
+    int hora;     
+    Paciente* paciente;
+public:
+    Atencion(string _fecha, int _hora, Paciente* paciente){
+        this->fecha = _fecha;
+        this->hora = _hora; 
+    }
+
+    string getFecha() { 
+        return fecha; 
+    }
+
+    virtual double calcularCostoConsulta() = 0;
+
+    virtual void mostrar() {
+        cout << "Fecha: " << fecha << ", Hora: " << hora << endl;
+        paciente->mostrarDatos();
+    }
+};
+
+class EnClinica : public Atencion {
```

Comment:
Como nombre de clase, no suena bien
```suggestion
class AtencionClinica : public Atencion {
```